pkg/runtime/framework/plugins/torch/torch.go

```diff
@@ -72,6 +74,39 @@ func (t *Torch) EnforceMLPolicy(info *runtime.Info, trainJob *trainer.TrainJob)
 		numProcPerNode = ptr.Deref(trainJob.Spec.Trainer.NumProcPerNode, intstr.FromString("auto"))
 	}
 
+	if jobTrainer := trainJob.Spec.Trainer; jobTrainer != nil && jobTrainer.ResourcesPerNode != nil {
+		var (
+			shouldUseCPU           func(resources corev1.ResourceList) bool
+			fallbackNumProcPerNode intstr.IntOrString
+		)
+		switch numProcPerNode.String() {
+		case "auto":
+			shouldUseCPU = func(resources corev1.ResourceList) bool {
+				for resName := range resources {
+					if strings.Contains(strings.ToLower(resName.String()), "gpu") {
+						return false
+					}
+				}
+				return true
+			}
+			fallbackNumProcPerNode = intstr.FromString("auto")
+		case "cpu":
+			shouldUseCPU = func(resources corev1.ResourceList) bool {
```

Comment:
```suggestion
			shouldUseCPU = func(corev1.ResourceList) bool {
```

nits: This is not used anywhere.

Reply:
Sure, I'll open another PR to address it later. Thanks for your Approval!